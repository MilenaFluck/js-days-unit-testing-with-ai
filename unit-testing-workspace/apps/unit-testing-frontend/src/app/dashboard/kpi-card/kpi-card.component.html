<mat-card class="dashboard-card">
  <mat-card-header class="dashboard-card__header">
    <mat-card-title>
      <h3 class="kpi-title">{{ title | uppercase }}</h3>
    </mat-card-title>
  </mat-card-header>
  <mat-card-content class="dashboard-card-content kpi-result">
    <span>{{ value }}</span>
    <p *ngIf="difference">{{ difference }}</p>
  </mat-card-content>
</mat-card>
