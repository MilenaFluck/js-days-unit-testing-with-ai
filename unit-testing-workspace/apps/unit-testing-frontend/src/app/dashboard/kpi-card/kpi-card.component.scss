@import "../../../assets/colors";

.dashboard-card {
  box-shadow: none;
  height: 160px;
  box-sizing: border-box;
}
.kpi-title {
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 18px;
  letter-spacing: 0.02em;
  text-align: left;
  color: map-get($app-secondary-palette, 400);
  margin: 0 -8px;
}
.kpi-result {
  font-size: 24px;
  font-style: normal;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: 0;
  text-align: left;
  color: #43436B;
  margin-top: 4px;

  p {
    font-size: 12px;
    font-style: normal;
    font-weight: 500;
    margin-top: 4px;
  }
}
