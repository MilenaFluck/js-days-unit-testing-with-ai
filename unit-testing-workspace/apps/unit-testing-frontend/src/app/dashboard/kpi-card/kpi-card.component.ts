import { Component, Input } from "@angular/core";

@Component({
  selector: 'app-kpi-card',
  templateUrl: './kpi-card.component.html',
  styleUrls: ['./kpi-card.component.scss']
})
export class KpiCardComponent {
  @Input() title!: string;
  @Input() value!: number;
  @Input() difference?: string;

}
