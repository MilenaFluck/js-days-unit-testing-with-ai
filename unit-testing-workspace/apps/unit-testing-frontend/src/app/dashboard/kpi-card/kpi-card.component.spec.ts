import { MatCardModule } from '@angular/material/card';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { KpiCardComponent } from './kpi-card.component';

describe('KpiCardComponent', () => {
  let kpiCardComponent: Spectator<KpiCardComponent>;

  const createComponent = createComponentFactory({
    component: KpiCardComponent,
    imports: [MatCardModule],
  });

  beforeEach(() => {
    kpiCardComponent = createComponent();
    kpiCardComponent.detectChanges();
  });

  it('should render kpi card without difference', () => {
    kpiCardComponent.component.title = 'Besucher';
    kpiCardComponent.component.value = 10000000;
    kpiCardComponent.detectChanges();

    expect(kpiCardComponent.element).toMatchSnapshot();
  });

  it('should render kpi card with difference', () => {
    kpiCardComponent.component.title = 'Besucher';
    kpiCardComponent.component.value = 10000000;
    kpiCardComponent.component.difference = '10%';
    kpiCardComponent.detectChanges();

    expect(kpiCardComponent.element).toMatchSnapshot();
  });

  it('should render kpi card with difference inline', () => {
    kpiCardComponent.component.title = 'Besucher';
    kpiCardComponent.component.value = 10000000;
    kpiCardComponent.component.difference = '10%';
    kpiCardComponent.detectChanges();

    expect(kpiCardComponent.element).toMatchInlineSnapshot();
  });
});
