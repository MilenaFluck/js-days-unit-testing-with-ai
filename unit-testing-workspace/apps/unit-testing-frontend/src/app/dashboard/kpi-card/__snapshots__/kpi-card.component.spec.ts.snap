// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KpiCardComponent should render kpi card with difference 1`] = `
<div
  id="root1"
>
  <mat-card
    class="mat-mdc-card mdc-card dashboard-card"
  >
    <mat-card-header
      class="mat-mdc-card-header dashboard-card__header"
    >
      <div
        class="mat-mdc-card-header-text"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          <h3
            class="kpi-title"
          >
            BESUCHER
          </h3>
        </mat-card-title>
      </div>
    </mat-card-header>
    <mat-card-content
      class="mat-mdc-card-content dashboard-card-content kpi-result"
    >
      <span>
        10000000
      </span>
      <p>
        10%
      </p>
      
    </mat-card-content>
  </mat-card>
</div>
`;

exports[`KpiCardComponent should render kpi card without difference 1`] = `
<div
  id="root0"
>
  <mat-card
    class="mat-mdc-card mdc-card dashboard-card"
  >
    <mat-card-header
      class="mat-mdc-card-header dashboard-card__header"
    >
      <div
        class="mat-mdc-card-header-text"
      >
        <mat-card-title
          class="mat-mdc-card-title"
        >
          <h3
            class="kpi-title"
          >
            BESUCHER
          </h3>
        </mat-card-title>
      </div>
    </mat-card-header>
    <mat-card-content
      class="mat-mdc-card-content dashboard-card-content kpi-result"
    >
      <span>
        10000000
      </span>
      
    </mat-card-content>
  </mat-card>
</div>
`;
